apollo/src/substrate2parachain/substrate2parachain.service.ts

```diff
@@ -0,0 +1,199 @@
+        toChain,
+        bridge: 'helix',
+      });
+
+      const latestNonce = firstRecord ? firstRecord.nonce : -1;
+
+      const res = await axios.post(url, {
+        query: `query { s2sEvents (first: ${this.fetchHistoryDataFirst}, orderBy: NONCE_ASC, filter: {nonce: {greaterThan: "${latestNonce}"}}) {totalCount nodes{id, laneId, nonce, amount, startTimestamp, endTimestamp, requestTxHash, responseTxHash, result, senderId, recipient, fee}}}`,
+        variables: null,
+      });
+
+      const nodes = res.data?.data?.s2sEvents?.nodes;
+      const timezone = new Date().getTimezoneOffset() * 60;
+
+      if (nodes && nodes.length > 0) {
+        for (const node of nodes) {
+          await this.aggregationService.createHistoryRecord({
+            id: `${keyPrefix}-${node.id}`,
+            fromChain,
+            toChain,
+            bridge: 'helix',
+            laneId: node.laneId,
+            nonce: global.BigInt(node.nonce),
+            requestTxHash: node.requestTxHash,
+            responseTxHash: node.responseTxHash,
+            sender: node.senderId,
+            recipient: node.recipient,
+            token: 'PRING',
+            amount: node.amount,
+            startTime: getUnixTime(new Date(node.startTimestamp)) - timezone,
+            endTime: getUnixTime(new Date(node.endTimestamp)) - timezone,
+            result: node.result,
+            fee: node.fee,
+            feeToken: this.lockFeeToken,
+          });
+
+          if (node.result == 0) {
+              if (!this.needSyncLockConfirmed && isLock) {
+                  this.needSyncLockConfirmed = true;
+              } else if (!this.needSyncBurnConfirmed && !isLock) {
+                  this.needSyncBurnConfirmed = true;
+              }
+          }
+        }
+
+        this.logger.log(
+          `save new ${fromChain} to ${toChain} records success, latestNonce: ${latestNonce}, added: ${nodes.length}`
+        );
+      }
+    } catch (e) {
+      this.logger.warn(
+        `fetch ${fromChain} to ${toChain} records failed ${e}`
+      );
+    }
+  }
+
+  async checkConfirmedLockRecords(isLock: boolean) {
+    if (!this.needSyncLockConfirmed && isLock) {
+      return;
+    } else if (!this.needSyncBurnConfirmed && !isLock) {
+      return;
+    }
+
+    const [fromChain, toChain, url, keyPrefix] = this.fetchInfos(isLock); 
+
+    try {
+      const { records: unconfirmedRecords } = await this.aggregationService.queryHistoryRecords({
+        take: this.fetchHistoryDataFirst,
+        where: {
+          fromChain,
+          toChain,
+          bridge: 'helix',
+          result: 0,
+        },
+      });
+
+      if (unconfirmedRecords.length == 0) {
+        if (isLock) {
+          this.needSyncLockConfirmed = false;
+        } else {
+          this.needSyncBurnConfirmed = false;
+        }
+        return;
+      }
+
+      const targetNonces = [];
+
+      for (const record of unconfirmedRecords) {
+        targetNonces.push('"' + record.nonce + '"');
```

Comment:
`targetNonces.push(record.nonce.toString());` ?

Reply:
this will put into subql request.